pkg/metrics/metrics_transformer.go

```diff
@@ -0,0 +1,220 @@
+package metrics
```

Comment:
Although it made sense to keep it under 'metrics', may be we should consider moving this to a different package.